tsconfig.json

```diff
@@ -20,16 +20,27 @@
     "strict": true,
     "strictNullChecks": true,
     "resolveJsonModule": true,
-    "jsx": "react"
+    "jsx": "react",
+    "paths": {
+      "generic-filehandle": ["./src/alias/generic-filehandle.ts"],
+      "zlib": ["./src/alias/zlib.ts"],
+      "vm": ["./src/alias/vm.ts"],
+      "@gmod/bgzf-filehandle": ["./src/alias/bgzf-filehandle.ts"],
+      "gosling.js": ["./src/index.ts"],
+      "@gosling.schema": ["./src/core/gosling.schema.ts"],
+      "@higlass.schema": ["./src/core/higlass.schema.ts"]
+    },
+    "baseUrl": "."
```

Comment:
This is the key to supporting the custom resolution. esbuild respects these paths when resolve module names, even in dependencies.

Reply:
Typescript will also respect the `gosling.js` & `@*` aliases with the editor source code. This allows us to point the compiler to the correct place when when typechecking the code, but ultimately `webpack.config.js` looks in `dist/gosling.es.js` for the entrypoint.

Reply:
As I am not that familiar with aliases, may I ask how you determine to select those aliases (e.g., zlib, vm, ...)?  I can only think of manually looking into `@gmod` codes and including any dependent modules to this list if they require implementation on our own for the esbuild.

Reply:
Yes this is what I had to do currently. Ideally very few dependencies for gosling include usage of node builtins. All the "problem" modules for bundling are isolated to `@gmod/*` and it took sometime to find the error boundaries. I can make some PRs to these dependencies. The use of `Proxy` in for the node-builtins makes it so that we get a runtime error if any other methods try to be accessed by other modules:

```
// alias/zlib.ts
export default new Proxy(
    {},
    {
        get(key) {
            if (key === 'inflateSync') return (src: Uint8Array) => Buffer.from(unzlibSync(src));
            throw new Error(
                'Module "zlib" has been externalized for browser compatibility and cannot be accessed in client code.'
            );
        }
    }
);
```

```javascript
import zlib from './alias/zlib.js';
zlib.inflateSync(...) // ok
zlib.flush() // Error! Module "zlib" has been exernalized for browser....
```

Reply:
https://github.com/GMOD/binary-parser/pull/23

Reply:
https://github.com/GMOD/bgzf-filehandle/pull/50

Reply:
https://github.com/GMOD/generic-filehandle/pull/88